### PR TITLE
Add Responsive Contentful Visual

### PR DESCRIPTION
**Pull request by sjstark**
Used responsive-craft-visual as base model for creating responsive-contentful-visual

### CODE REVIEW

**Comment by weotch** [comment]
Awesome!!

Are you currently using this on sxm?  If so, can you point me to where?  I'd like to DRY the shared stuff between this and `responsive-craft-visual` into a `responsive-cloak-visual` component.

File: components/responsive-contentful-visual.vue [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	destroyed: ->
+		@isLanscapeMediaQuery?.removeListener @checkIsLandscape
+	computed:
+		# Visual configs
+		landscape: -> @makeConfig @getAsset('image', 'landscape'),
+			@getAsset('video', 'landscape')
+		portrait: -> @makeConfig @getAsset('image', 'portrait'),
+			@getAsset('video', 'portrait')
+		# Do we have unique landscape and portrait configs?
+		isResponsive: -> !!(@landscape and @portrait)
```
Comment (weotch) [inline components/responsive-contentful-visual.vue:36]: Can you restore the newlines between code blocks like this?

```suggestion
		# Visual configs
		landscape: -> @makeConfig @getAsset('image', 'landscape'),
			@getAsset('video', 'landscape')
		portrait: -> @makeConfig @getAsset('image', 'portrait'),
			@getAsset('video', 'portrait')
			
		# Do we have unique landscape and portrait configs?
		isResponsive: -> !!(@landscape and @portrait)
```

Something about how you copied the source from https://github.com/BKWLD/cloak/blob/master/components/responsive-craft-visual.vue destroyed all those newlines.

**Comment by sjstark** [comment]
> Awesome!!
> 
> Are you currently using this on sxm? If so, can you point me to where? I'd like to DRY the shared stuff between this and `responsive-craft-visual` into a `responsive-cloak-visual` component.

I'm working on breaking out the section that used it into its own branch, I realized I had quite a bit being added to one working branch and that it'll be easier to track with each feature on its own branch.

However, Jon mentioned here that he already has it running in his own branch as well. https://gitlab.com/sxm-media/sxm-media/-/merge_requests/3#note_601245290

**Comment by weotch** [comment]
@sjstark FYI i ended up going with a mixin approach rather than a functional component adapter.  I don't _love_ this, I could see rewriting it later.  Here's some notes:

 https://github.com/BKWLD/cloak/blob/4ba1a73d3a644b31e050b4f8518741f7ac6cdf2f/mixins/visual/responsive-visual.vue#L1-L4

**Pull request merged**